src/dsp_tools/commands/xmlupload/iiif_client.py

```diff
@@ -0,0 +1,61 @@
+from dataclasses import dataclass
+from typing import Protocol
+
+import requests
+
+from dsp_tools.commands.xmlupload.models.input_problems import IIIFUriProblem
+
+
+@dataclass(frozen=True)
+class IIIFUriValidator(Protocol):
+    """Interface (protocol) for communication with external IIIF-servers to do a health check."""
+
+    uri: str
+    passed_regex: bool
```

Comment:
This naming is confusing, because it makes me assume that the thing stored in this field is a regex, and that "passed" is a modifier for it, so it makes me think that this is a regex that has passed.

What you actually want to store is the information if the regex has passed or not. There is a best practice for naming bool variables: they should ideally start with "has" or "is", i.e. they answer the question "Has the regex passed?"

In this case, the name could be: 

```suggestion
    has_regex_passed: bool
```

or: 

```suggestion
    regex_has_passed: bool
```